app/ui/src/app/integration/detail-page/detail.component.ts

```diff
@@ -129,7 +131,20 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
       });
   }
 
-  deploymentAction(event, deployment) {
+  draftAction(event) {
+    switch (event.id) {
+      case 'publish':
+        this.integrationActionsService.requestAction('publish', <any> this.integration);
+        break;
+      case 'edit':
+        this.integrationActionsService.requestAction('edit', <any> this.integration);
+        break;
+      default:
+        break;
+    }
+  }
+
+  deploymentAction(event, deployment: IntegrationDeployment) {
     switch (event.id) {
       case REPLACE_DRAFT:
```

Comment:
@chirino baffled how this is supposed to work?  Previously deployments had steps in them and I copied them into the integration object and saved it but...

Reply:
a Deployment's spec field hold an Integration object the can replace the current integration.

Reply:
Make sure your working with a whole deployment and not a deployment overview.